src/hpo_service.py

```diff
@@ -33,7 +36,7 @@ def __init__(self):
     def newExperiment(self, id_, experiment_name, total_trials, parallel_trials, direction, hpo_algo_impl,
                       objective_function, tunables, value_type):
         if self.containsExperiment(experiment_name):
-            print("Experiment already exists")
+            logger.error(HPOErrorConstants.EXPERIMENT_EXISTS)
```

Comment:
It looks like we are not returning a error value here in `newExperiment` and the calling function goes ahead even if this fails. In addition to the logging the error message we need to return a error value which needs to be checked from the calling function.

Reply:
We are already validating when the request is received and return the response initially only. So this check isn't required again. Removed this and added the pre-check in gRPC as well